### PR TITLE
Octave: fix java issues

### PR DESCRIPTION
**Pull request by schoeps**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

see https://github.com/Homebrew/homebrew-core/issues/16979

### CODE REVIEW

File: Formula/octave.rb [hunk 1]
```diff
@@ -63,13 +63,22 @@ def install
...
     end
 
     # Default configuration passes all linker flags to mkoctfile, to be
     # inserted into every oct/mex build. This is unnecessary and can cause
     # cause linking problems.
     inreplace "src/mkoctfile.in.cc", /%OCTAVE_CONF_OCT(AVE)?_LINK_(DEPS|OPTS)%/, '""'
 
+    # allow for recent Oracle Java (>=1.8) without requiring the old Apple Java 1.6
+    # this is more or less the same as in http://savannah.gnu.org/patch/index.php?9439
+    inreplace "libinterp/octave-value/ov-java.cc",
```
Comment (ilovezfs) [inline Formula/octave.rb]: This inreplace and the one below are still needed for head?

**Comment by schoeps** [comment]
Sadly yes. It takes much longer for Mac related patches to be accepted (the other patch is also needed for Linux and Windows users, so it was quicker).

**Comment by schoeps** [comment]
sorry. closed by accident

**Comment by ilovezfs** [comment]
Thanks @schoeps! Shipped!

**Pull request closed without merging**